shared/reducers/chat.js

```diff
@@ -79,6 +79,16 @@ function reducer (state: State = initialState, action: Actions) {
   switch (action.type) {
     case CommonConstants.resetStore:
       return initialState
+    case 'chat:removeOutboxMessage': {
```

Comment:
The deletedIDs set is for MessageIDs, and these outbox messages don't have MessageIDs.  So we'd have to expand it to take outboxIDs too.

But I don't know whether we should.  I think we're supposed to expect that the RPC completing means that the message is gone and will never be sent to us in a message load again.  If it ever was, that sounds it would be a big privacy problem, and it might be better for us to learn about it by seeing a deleted message reappear rather than papering over it with our deletedIDs set.  @mmaxim, any thoughts?